File: backend/src/main/java/com/cybercrime/model/Complaint.java
package com.cybercrime.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "complaints")
public class Complaint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @ManyToOne(fetch = FetchType.LAZY)
    private Department department;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    private String location;
    private LocalDateTime incidentDate;
    private LocalDateTime createdAt;
    private LocalDateTime resolvedDate;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "priority")
    private ComplaintPriority priority;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private ComplaintStatus status;
    
    private String suspectInfo;
    private String suspectSocialMedia;
    private String suspectPhoneNumber;
    private Integer rating;
    private String feedback;
    
    @Column(name = "feedback_date")
    private LocalDateTime feedbackDate;
    
    @Column(columnDefinition = "TEXT")
    private String resolutionNotes;
    
    @OneToMany(mappedBy = "complaint", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Evidence> evidences;
}

File: frontend/src/types/complaint.ts
import { Evidence, Feedback, User } from "./user";

export type ReporterType = 'victim' | 'reporter' | 'business' | 'witness';
export type ComplaintStatus = 'PENDING' | 'IN_PROGRESS' | 'RESOLVED' | 'REJECTED';
export type ComplaintPriority = 'LOW' | 'MEDIUM' | 'HIGH';

export interface ComplaintCreateDto {
  title: string;
  description: string;
  departmentId: string | number;
  location: string;
  incidentDate: string;
  suspectInfo?: string;
  suspectSocialMedia?: string;
  suspectPhoneNumber?: string;
}

export interface Department {
  id: number;
  name: string;
}

export interface Complaint extends Omit<ComplaintCreateDto, 'departmentId'> {
  id: number;
  status: ComplaintStatus;
  createdAt: string;
  resolvedDate?: string;
  userId: number;
  user?: User;
  departmentId?: number;
  department?: Department;
  evidences?: Evidence[];
  feedback?: Feedback;
}

export interface ComplaintDto {
    id: number;
    title: string;
    description: string;
    status: ComplaintStatus;
    priority: ComplaintPriority;
    location: string;
    incidentDate: string;
    createdAt: string;
    updatedAt?: string;
    suspectInfo?: string;
    suspectSocialMedia?: string;
    suspectPhoneNumber?: string;
    rating?: number;
    feedback?: string;
    feedbackDate?: string;
    evidences?: Array<{
        id: number;
        filePath: string;
        fileType: string;
    }>;
    user: {
        id: number;
        name: string;
        email: string;
    };
    department?: {
        id: number;
        name: string;
    };
}